java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/NodeActionsProvider.java

```diff
@@ -161,4 +166,40 @@ public static NodeActionsProvider forFile(FileObject f) {
         return CompletableFuture.completedFuture(true);
     }
     
+    CompletableFuture<Object> invokeAction(NbCodeLanguageClient client, String category, String aid, List<Object> arguments) {
+        String path = "Actions/" + category + "/" + aid.replace('.', '-') + ".instance"; //NOI18N
+        FileObject config = FileUtil.getConfigFile(path);
+        String contextType = (String) config.getAttribute("type"); //NOI18N
+        try {
+            Class<?> clazz = Thread.currentThread().getContextClassLoader().loadClass(contextType);
+            Object context = gson.fromJson(gson.toJson(arguments.get(0)), clazz);
```

Comment:
Q: this deserializes the 1st argument, which was deserialized at line ~97 as JSONObject, in an attempt to read inputs like `{ data : { id: <nodeId> } }`. Does the code that deserializes a JSONObject / id behave "well" when it is fed with a typed parameter (other node-based actions) ?

Reply:
In case the new method `invokeAction()` is called the first argument doesn't contain input you mention above (`data`) as it is not called from tree view. It is expected to contain fields for a action type to deserialise properly